### PR TITLE
BEMHTML: Should not render String or Array as attrs value

### LINKED ISSUE #187: BEMHTML: Should not render String or Array as attrs value
```js
  it('should not return String as attrs value', function() {
    test(function() {},
      { block: 'b', attrs: 'skipme' },
      '<div class="b"></div>');
  });

  it('should not return Array as attrs value', function() {
    test(function() {},
      { block: 'b', attrs: [ 1, 2 ] },
      '<div class="b"></div>');
  });
```

### PR DESCRIPTION
**Pull request by miripiruni**
Fix #187 

Benchmarks is OK.

### CODE REVIEW

File: lib/bemhtml/index.js [hunk 1]
```diff
@@ -158,7 +158,7 @@ BEMHTML.prototype.renderClose = function renderClose(prefix,
 
   // NOTE: maybe we need to make an array for quicker serialization
   attrs = utils.extend(attrs, ctx.attrs);
-  if (attrs) {
+  if (attrs && Object.prototype.toString.call(attrs) === '[object Object]') {
```
Comment (zxqfox) [inline lib/bemhtml/index.js]: `toString = Function.prototype.call.bind(Object.prototype.toString)` ?

**Comment by zxqfox** [comment]
👍

File: lib/bemhtml/index.js (hunk 1 above)
Comment (miripiruni) [inline lib/bemhtml/index.js]: @zxqfox IMHO it’s too much… Why I should do this?

**Comment by miripiruni** [comment]
Review needed.

cc @tadatuta @veged

File: lib/bemhtml/index.js (hunk 1 above)
Comment (zxqfox) [inline lib/bemhtml/index.js]: Because you already did `Object.prototype.toString.call(attrs) === '[object Object]'` instead of `typeof attrs === 'object'`

Comment (miripiruni) [inline lib/bemhtml/index.js]: @zxqfox OK, I’ll rewrite it in more explicit way.

File: lib/bemhtml/index.js [hunk 2]
```diff
@@ -158,7 +158,8 @@ BEMHTML.prototype.renderClose = function renderClose(prefix,
 
   // NOTE: maybe we need to make an array for quicker serialization
   attrs = utils.extend(attrs, ctx.attrs);
-  if (attrs) {
+  if (attrs && typeof attrs === 'object' && !Array.isArray(attrs) &&
+      attrs !== null) {
     var name; // TODO: do something with OmetaJS and YUI Compressor
```
Comment (veged) [inline lib/bemhtml/index.js]: obsolete TODO ;-)

**Comment by veged** [comment]
LGTM

File: lib/bemhtml/index.js (hunk 2 above)
Comment (miripiruni) [inline lib/bemhtml/index.js]: @veged TODO removed.

File: lib/bemhtml/index.js [hunk 3]
```diff
@@ -158,8 +158,9 @@ BEMHTML.prototype.renderClose = function renderClose(prefix,
 
   // NOTE: maybe we need to make an array for quicker serialization
   attrs = utils.extend(attrs, ctx.attrs);
-  if (attrs) {
-    var name; // TODO: do something with OmetaJS and YUI Compressor
+  if (attrs && typeof attrs === 'object' && !Array.isArray(attrs) &&
+      attrs !== null) {
+    var name;
     /* jshint forin : false */
     for (name in attrs) {
```
Comment (veged) [inline lib/bemhtml/index.js]: should be `for (var name in attrs) {`

also, as I remember, there was one more similar TODO somewhere

Comment (miripiruni) [inline lib/bemhtml/index.js]: > should be for (var name in attrs) {

Updated.

> also, as I remember, there was one more similar TODO somewhere

I can’t find…

**Pull request merged**